src/en/buttsmithy/src/eu/kanade/tachiyomi/extension/en/buttsmithy/Buttsmithy.kt

```diff
@@ -0,0 +1,308 @@
+            else -> chapTitle.replace(" ", "-").replace(".", "-")
+        }
+    }
+
+    private fun convertMenuElementToSManga(menuElement: Elements): List<SManga> {
+        val comicLinkSelector = ".menu-item-type-custom a[href]"
+
+        val linkElements = menuElement.select(comicLinkSelector)
+        return linkElements
+            // filter out the first Alfie chapter that is still hosted under "incase.buttsmithy.com"
+            // see "fetchAlfieSMangas()" for how Alfie should be retrieved
+            .filter { linkElement -> !linkElement.text().contains(alfieTitle) }
+            .map { linkElement ->
+                val comicTitle = linkElement.text()
+                val comicUrl = linkElement.attr("href")
+
+                SManga.create().apply {
+                    setUrlWithoutDomain(comicUrl)
+                    title = comicTitle
+                    author = inCase
+                    artist = inCase
+                    status = SManga.COMPLETED
+                    genre = "NSFW"
+                    thumbnail_url = generateImageUrlWithText(comicTitle)
+                    status = SManga.COMPLETED
+                }
+            }
+    }
+
+    private fun generateImageUrlWithText(text: String): String {
+        return "https://fakeimg.pl/800x1236/?text=$text&font=lobster"
+    }
+
+    private fun generateMangasPage(): MangasPage {
+        return MangasPage(fetchAllComics(), false)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> = throw Exception("Not used")
+
+    override fun imageUrlParse(response: Response): String {
+        val pageDoc = response.asJsoup()
+        return pageDoc.select("#comic").select("img[src]").attr("href")
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage = throw Exception("Not used")
+
+    override fun latestUpdatesRequest(page: Int): Request = throw Exception("Not used")
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
+        return Observable.just(
+            if (manga.title.contains(alfieTitle)) {
+
+                val pageDoc = client.newCall(GET(baseUrlAlfie, headers)).execute().asJsoup()
+                val mostRecentChapTitle = extractChapterTitleFromPageDoc(pageDoc)
+                val chapTitle = manga.title.substringAfter("Alfie - ").trim()
+
+                SManga.create().apply {
+                    url = "$chapterOverviewBaseUrl/${chapterTitleToChapterUrlName(chapTitle)}"
+                    title = "$alfieTitle - $chapTitle"
+                    author = inCase
+                    artist = inCase
+                    status = decideAlfieStatusFromTitle(chapTitle, mostRecentChapTitle)
+                    genre = "fantasy, NSFW"
+                    thumbnail_url = generateImageUrlWithText(alfieTitle)
+                }
+            } else manga
+        )
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request = throw Exception("Not used")
```

Comment:
This is always used by the webview. (tachiyomiorg/tachiyomi#1755)

```suggestion
    override fun mangaDetailsRequest(manga: SManga) =
        if (manga.title.contains(alfieTitle)) GET(manga.url) else GET(baseUrl + manga.Url)
    
```